presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -257,6 +270,13 @@ public NodeSelector createNodeSelector(Session session, ConnectorId connectorId,
                     .map(InternalNode::getNodeIdentifier)
                     .collect(toImmutableSet());
 
+            int weight = (int) ceil(1.0 * minVirtualNodeCount / activeNodes.size());
```

Comment:
Where is the logic of compute the actual virtual node count based on minVirtualNodeCount  and physical node count?

Reply:
"weight" is basically the number of virtual nodes created for each physical node. So the actual number of virtual node is `weight * physicalNodeCount = ceil(minVirtualNodeCount / physicalNodeCount) * physicalNodeCount >= minVirtualNodeCount`. I figured it's better to make the number of virtual node configurable, but then the exact number is probably not important, mostly just the order of magnitudes. Directly configuring the weight is another option, but then the desired weight could be very different for a 10 node cluster vs a 500 node cluster.

Reply:
Why don't we directly change the config to `node-scheduler.consistent-hashing-virtual-node-count` to represent the weight here? 
so it would be an independent config from how many nodes a cluster have

Reply:
Because, as I mentioned above, if you set weight directly, the actual number of virtual nodes on a hashing ring will depend on the number of physical node, which means it's hard to give a good default weight setting.